### PR TITLE
acl: remove initializeLegacyACL and the rest of the legacy FSM commands

### PR DESCRIPTION
**Pull request by dnephin**
Branched from #11101

This PR removes:
*  `initializeLegaclACL` from the leader initialization
*  a state store function that is no longer used
* the rest of the legacy FSM command and `ACLOp`
* renames a few things with a deprecated prefix to indicate they are no longer used, but must remain

### CODE REVIEW

File: agent/consul/fsm/snapshot_oss.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ func init() {
 	registerRestorer(structs.KVSRequestType, restoreKV)
 	registerRestorer(structs.TombstoneRequestType, restoreTombstone)
 	registerRestorer(structs.SessionRequestType, restoreSession)
-	registerRestorer(structs.ACLRequestType, restoreACL)
+	registerRestorer(structs.DeprecatedACLRequestType, restoreACL)
```
Comment (rboyer) [inline agent/consul/fsm/snapshot_oss.go:18]: Did you want to also clean up `restoreACL` yet?

Comment (dnephin) [reply agent/consul/fsm/snapshot_oss.go:18]: Don't we still need `restoreACL` for snapshots that contain legacy ACL entries? I kept the test around to show we can still restore snapshots with this data.

Comment (rboyer) [reply agent/consul/fsm/snapshot_oss.go:18]: Ah, right. I wasn't sure how much we were just going to restore old ACL data directly into `/dev/null` or not.

**Review by rboyer — APPROVED** [review]
LGTM

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/460777.

**Pull request merged**